tests/secret-providers/aws_secretmanager/aws_secretmanager_test.go

```diff
@@ -257,11 +283,54 @@ spec:
 )
 
 func TestAwsSecretManager(t *testing.T) {
+	var useJSONSecretFormat = false
+	require.NotEmpty(t, awsAccessKeyID, "TF_AWS_ACCESS_KEY env variable is required for AWS Secret Manager test")
+	require.NotEmpty(t, awsSecretAccessKey, "TF_AWS_SECRET_KEY env variable is required for AWS Secret Manager test")
+
+	// Create the secret in AWS
+	err := createAWSSecret(t, useJSONSecretFormat)
+	assert.NoErrorf(t, err, "cannot create AWS Secret Manager secret - %s", err)
+
+	// Create kubernetes resources for PostgreSQL server
+	kc := GetKubernetesClient(t)
+	data, postgreSQLtemplates := getPostgreSQLTemplateData()
+
+	CreateKubernetesResources(t, kc, testNamespace, data, postgreSQLtemplates)
+
+	assert.True(t, WaitForStatefulsetReplicaReadyCount(t, kc, postgreSQLStatefulSetName, testNamespace, 1, 60, 3),
+		"replica count should be %d after 3 minutes", 1)
+
+	createTableSQL := "CREATE TABLE task_instance (id serial PRIMARY KEY,state VARCHAR(10));"
+	psqlCreateTableCmd := fmt.Sprintf("psql -U %s -d %s -c \"%s\"", postgreSQLUsername, postgreSQLDatabase, createTableSQL)
+
+	ok, out, errOut, err := WaitForSuccessfulExecCommandOnSpecificPod(t, postgresqlPodName, testNamespace, psqlCreateTableCmd, 60, 3)
+	assert.True(t, ok, "executing a command on PostreSQL Pod should work; Output: %s, ErrorOutput: %s, Error: %s", out, errOut, err)
+
+	// Create kubernetes resources for testing
+	data, templates := getTemplateData(useJSONSecretFormat)
+
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 3),
+		"replica count should be %d after 3 minutes", minReplicaCount)
+
+	testScaleOut(t, kc, data)
+
+	// cleanup
+	KubectlDeleteMultipleWithTemplate(t, data, templates)
+	DeleteKubernetesResources(t, testNamespace, data, postgreSQLtemplates)
+
+	// Delete the secret in AWS
+	err = deleteAWSSecret(t)
+	assert.NoErrorf(t, err, "cannot delete AWS Secret Manager secret - %s", err)
+}
```

Comment:
Could we create a function for the test and pass the required changes? 'TestAwsSecretManager' and 'TestAwsSecretManagerJSONFormat' are almost the same, maybe we can create a function like `testAwsSecretManager` with the changes as arguments (useJSONSecretFormat , the templates, etc)

Reply:
Do you want testAwsSecretManager refactored? Or a new function?

Reply:
My proposal is to create a new function that can be called with both configurations to reduce the amount of duplicated code, the name was tentative and just a suggestion

Reply:
Ok thanks.